pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/TopicSchema.java

```diff
@@ -126,30 +144,63 @@ private static SchemaType getDefaultSchemaType(Class<?> clazz) {
     @SuppressWarnings("unchecked")
     private static <T> Schema<T> newSchemaInstance(Class<T> clazz, SchemaType type) {
```

Comment:
I would actually moving this method and `AutoConsumeSchema#getSchema(SchemaInfo schemaInfo)` to `Schema.java`. so that we have one central place for retrieving schema instances by schema type. Otherwise we are spreading this method across multiple modules in the project. things will become inconsistent as time goes. especially when a new schema type is introduced.